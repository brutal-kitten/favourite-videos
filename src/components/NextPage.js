import React, { Component } from 'react';


class NextPage extends Component {

  handleChange = (event) => {

    event.preventDefault();
    console.log("nextPage")
    let newPage = parseInt(this.props.currentPage, 10) + 1;
    if (newPage <= parseInt(this.props.totalPages, 10)) {
      this.props.setStartIndex(newPage);
    }
  }

  render () {

    return (
      <div className="nextPage">
        <button onClick={(event) => this.handleChange(event)}>
          <span className="glyphicon glyphicon-chevron-right"></span>
        </button>
      </div>
    )
  }
}

export default NextPage
