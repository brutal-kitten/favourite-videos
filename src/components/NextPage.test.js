import React from 'react';
import ReactDOM from 'react-dom';
import NextPage from './NextPage.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <NextPage />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should call setStartIndex with value of next page if there is next page', () => {
  const MockSetStartIndex = jest.fn();
  const props = {
    setStartIndex: MockSetStartIndex,
    totalPages: 2,
    currentPage: 1
  }

  const wrapper = shallow(<NextPage {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {},
  });

  expect(MockSetStartIndex.mock.calls.length).toBe(1);
  expect(MockSetStartIndex.mock.calls[0][0]).toBe(2);
});

it('should not call setStartIndex if there is no next page', () => {

  const MockSetStartIndex2 = jest.fn();
  const props2 = {
    setStartIndex: MockSetStartIndex2,
    totalPages: 2,
    currentPage: 2
  }
  
  const wrapper2 = shallow(<NextPage {...props2}/>);

  wrapper2.find('button').simulate('click', {
    preventDefault: () => {},
  });

  expect(MockSetStartIndex2.mock.calls.length).toBe(0);
})
