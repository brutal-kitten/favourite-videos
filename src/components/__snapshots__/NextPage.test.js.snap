// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="nextPage"
>
  <button
    className="buttonNextPage"
    onClick={[Function]}
    tabIndex="0"
  >
    <span
      className="glyphicon glyphicon-chevron-right"
    />
  </button>
</div>
`;
